### PR TITLE
Add warning about upcoming expanded fields limit

### PR DESCRIPTION
**Pull request by cbuescher**
Starting with Elasticsearch 7.0, we limit the number of automatically expanded
fields for the "all fields" mode ("default_field": "*") for the query_string and
simple_query_string queries to 1024 fields (see #26541). This change adds a
deprecation warning to the QueryParserHelper where in the next major version we
will throw an error to warn users of the upcoming change.

Relates to #26541

### CODE REVIEW

**Comment by cbuescher** [comment]
@jimczi this is meant as the follow up to yesterdays docs change (#34801). I think we also need some not in the migration docs, but as far as I understand I would need to add the note about the limit coming into effect on master, right? If so, I will do that in another PR, if I should add this to some 6.x migration doc please let me know in this PR. I always get confused where these notes should go.

**Review by jimczi — APPROVED** [review]
LGTM, thanks @cbuescher

**Review by jimczi — COMMENTED** [review]
You can add a deprecation note in https://github.com/elastic/elasticsearch/blob/6.x/docs/reference/migration/migrate_6_0/search.asciidoc

**Comment by cbuescher** [comment]
> You can add a deprecation note in https://github.com/elastic/elasticsearch/blob/6.x/docs/reference/migration/migrate_6_0/search.asciidoc

Isn't that document intended for migrations _to_ 6.0? My understanding so far was that we add breaking changes in the versions that they should be adressed. That would be 7.0 for the limit introduced here, at least for the breaking change. https://github.com/elastic/elasticsearch/tree/6.x/docs/reference/migration/migrate_7_0 looks quite empty though, so I guess somewhere in https://github.com/elastic/elasticsearch/tree/master/docs/reference/migration/migrate_7_0 would be the right spot? 
I will go ahead and add a deprecation note to https://github.com/elastic/elasticsearch/blob/6.x/docs/reference/migration/migrate_6_5.asciidoc#breaking_65_search_changes in this PR.

**Comment by cbuescher** [comment]
I added deprecation notes in a new `migrate_6_6.asciidoc` in the last commit, I think its okay to only backport this change to the 6.x branch. Let me know if you disagree, otherwise I'll just merge to 6.x. I'll open a separate PR for the 7.0 migration notes.

File: docs/reference/migration/migrate_6_6.asciidoc [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+[float]
+[[breaking_66_search_changes]]
+=== Search changes
+
+[float]
+==== `query_string` and `simple_query_string` query
+
+Using automatically expanded fields for the "all fields" mode ("default_field": "*") 
+for the `query_string` and `simple_query_string` now raises a warning and a deprecation
```
Comment (jimczi) [inline docs/reference/migration/migrate_6_6.asciidoc]: it is also applied to `multi_match` queries.

**Review by jimczi — APPROVED** [review]
Yep +1 for 6x only.

**Pull request merged**